src/app/file-manager/file-explorer/file-explorer.component.ts

```diff
@@ -213,13 +211,19 @@ export class FileExplorerComponent implements OnInit {
   }
 
   dataFinishedLoading() {
-    let dataFinishedLoading = false;
-    if (this.isHome && this.firecloudService.getIsResponseComplete()) {
-      dataFinishedLoading = true;
-    } else if (this.bucketService.getIsResponseComplete()) {
-      dataFinishedLoading = true;
-    }
-    return dataFinishedLoading;
+    return this.firecloudService.getIsResponseComplete() || this.bucketService.getIsResponseComplete();
```

Comment:
I think this is `&&` not `||` since both must be complete.

Reply:
Both must not be completed at the same time, if we are in the first page we only call firecloud to get workspaces data, if we move to another folder we call bucket service to get bucket info but this occur in different moments. I just added isHome condition to check if we need to verify one or another response

Reply:
OK, as long as we know which one the UI is currently waiting for.